### ISSUE TITLE
CustomizedRuntime-4.cpp:105:2: error: expected unqualified-id

### ISSUE THREAD

**Issue by neulxf** [tancheng/mlir-cgra#3]
Hello!
When I follow the instructions in the README and run the demo, there are no errors. However, when testing MiniLM and running sh script4x4.sh, the following error occurs.
![image](https://github.com/tancheng/mlir-cgra/assets/100358779/f5850fda-5ab2-484b-b941-4e94818c9ee0)
![image](https://github.com/tancheng/mlir-cgra/assets/100358779/03cd6edc-790d-4485-8d42-a4bebfce5a9f)

**Comment by tancheng** [comment]
Hi neulxf, seems the lib is not linked well. Did you use [this](https://github.com/tancheng/mlir-cgra/blob/49ee1c981a3d1fba9450792b93f6d7c6ab660670/experiments/MiniLM/cgra/script4x4.sh#L38C47-L38C107) to compile.

**Comment by neulxf** [comment]
Yes, I used the given docker.  I found the `mlir-cgra`is old, so I executed the `git pull` in `mlir-cgra`
I did not change the source code, and did not rebuild anything

**Comment by tancheng** [comment]
How about the other tests? the one inside the demo folder? Give me some time, I will also try it out.

**Comment by neulxf** [comment]
Thanks! The Demo is success, MiniLM is wrong, and Sentence-BERT spent hours with no results.

**Comment by tancheng** [comment]
BTW, docker might be slow with limited ram provided. Though the local installation is error-prone in terms of versions, I still recommend that. Anyways, I will try the docker.

**Comment by neulxf** [comment]
Thanks for the suggestion,，I'm trying to run it on a physical machine



---- Replied Message ----
| From | Cheng ***@***.***> |
| Date | 12/09/2023 15:00 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [tancheng/mlir-cgra] CustomizedRuntime-4.cpp:105:2: error: expected unqualified-id (Issue #3) |

BTW, docker might be slow with limited ram provided. Though the local installation is error-prone in terms of versions, I still recommend that. Anyways, I will try the docker.

**Comment by neulxf** [comment]
I built the project by myself but met the same error.
I pass the error by adding `-L`  in front of `../../../../llvm-project/build/lib/libmlir_c_runner_utils.so`.
![image](https://github.com/tancheng/mlir-cgra/assets/100358779/92afb738-8d23-46b3-8fd8-6814aae0254f)
But occur new error:
![image](https://github.com/tancheng/mlir-cgra/assets/100358779/8330b81f-9090-4875-a618-4aa4a6693bd5)

**Comment by neulxf** [comment]
Thanks！ I passed the error， I found the miss `mlir_c_runner_utils.dir/CRunnerUtils.cpp.o` file, and added the path to `script4x4.sh`

```sh
clang++-12 main.cpp 11-model-4.o 12-accel-4.o /home/workspace/tools/llvm-project/build/tools/mlir/lib/ExecutionEngine/CMakeFiles/mlir_c_runner_utils.dir/CRunnerUtils.cpp.o -L../../../../llvm-project/build/lib/libmlir_c_runner_utils.so  CustomizedRuntime-4.cpp -I../../../sim/ ../../../sim/*.cpp -o simulate4

```
And get the result. It looks like it should be right.
![image](https://github.com/tancheng/mlir-cgra/assets/100358779/4fad6910-c346-4ff4-bb50-1ff086188cb7)

**Comment by tancheng** [comment]
Hi neulxf,

Thanks for the fix. I did remember I have some comments in one of the script mentioning we need to set some env variable to include that path. But I can't find them in any of the script now... Let's leave this issue open to help the others.

Thank you!
